src/common/quantile.cc

```diff
@@ -390,7 +396,7 @@ void SketchContainerImpl<WQSketch>::MakeCuts(HistogramCuts* cuts) {
   cuts->min_vals_.HostVector().resize(sketches_.size(), 0.0f);
   std::vector<typename WQSketch::SummaryContainer> final_summaries(reduced.size());
 
-  ParallelFor(reduced.size(), n_threads_, Sched::Guided(), [&](size_t fidx) {
```

Comment:
It's weird, the guided schedule is causing those test failures. Switching to auto fixes them. Maybe some bug in OpenMP?

Reply:
Hmm, never mind, looks like auto also fails sometimes. Probably some kind of race condition in this parallel for.

Reply:
Ok, finally found the root cause.